code/datums/bug_report.dm

```diff
@@ -0,0 +1,183 @@
+// Datum for handling bug reports
+#define STATUS_SUCCESS 201
+
+/datum/tgui_bug_report_form
+	// contains all the body text for the bug report.
+	var/list/bug_report_data = null
+
+	// client of user who created the initial report, immutable, set on init.
+	var/client/initial_user = null
+
+	// client of the admin who is accessing the report, we don't want multiple admins unknowingly making changes at the same time.
+	var/client/admin_user = null
+
+	// value to determine if the bug report is submitted and awaiting admin approval, used for state purposes in tgui.
+	var/awaiting_admin_approval = FALSE
+
+	// for garbage collection purposes.
+	var/selected_confirm = FALSE
+
+/datum/tgui_bug_report_form/New(mob/user)
+	if(!user.client) // nope.
+		return INITIALIZE_HINT_QDEL
```

Comment:
You can only do this in Initialize. As I mentioned its fine to set the initial_user here in New but then in Initialize check if initial_user is valid. if its not, then return `INITIALIZE_HINT_QDEL`

Reply:
INITIALIZE_HINT_QDEL is for atoms, it's part of the atom subsystem.